src/main/java/com/bajdas/rest_shop/product/ProductController.java

```diff
@@ -0,0 +1,38 @@
+package com.bajdas.rest_shop.product;
+
+import com.bajdas.rest_shop.model.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+public class ProductController {
+
+  private ProductFinder productFinder;
+
+  @Autowired
+  public ProductController(ProductFinder productFinder) {
+    this.productFinder = productFinder;
+  }
+
+  @GetMapping("/productInfo")
+  public ResponseEntity<Product> getProductInfo(@RequestParam(required = false) String name,
```

Comment:
Jak masz ID to nie potrzebujesz name :)

Reply:
A, juz rozumiem. Ale to raczej robisz dwa osobne entrypointy - getProductById, findProductByName. Szczegolnie, ze by name niekoniecznie musi zwracac jeden wynik